src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -43,6 +97,22 @@ public LogicalLocation TransformLogicalLocationVersionOne(LogicalLocationVersion
             return logicalLocation;
         }
 
+        public Hash TransformHashVersionOne(HashVersionOne node)
+        {
+            Hash hash = null;
+
+            if (node != null)
+            {
+                hash = new Hash
+                {
+                    Algorithm = (AlgorithmKind)Enum.Parse(typeof(AlgorithmKind), node.Algorithm.ToString()),
```

Comment:
1. In v2, `hash.algorithm` is a `string`. I think you didn't remove the `EnumHint` from `Hash.Algorithm`. Once you've done that, it's just `Algorithm = node.Algorithm.ToString();`
2. **BUT**! You can't just copy the string, because the spec says that if your algorithm is one of those that appears in the IANA registry of [Hash function textual names](https://www.iana.org/assignments/hash-function-text-names/hash-function-text-names.xhtml), then you have to spell it like IANA spells it. For example, we have `AlgorithmKind.Sha256`, but IANA spells it `"sha-256"` (note the hyphen). Don't worry, there are only 7 hash functions in the IANA registry.

3. Although the spec says that `hash.algorithm` is case-insensitive, I suggest you pull the values to lower case for consistency with IANA.

Reply:
I had the nagging sense that this wasn't an enum anymore...

Reply:
I see you made the change but you didn't change the spellings. It needs to be `"sha-1"`, "`sha-256"` to match the IANA registry (says the spec).